### PR TITLE
Configured VAProfileRedis::V2::ContactInformation

### PR DESCRIPTION
**Pull request by RachalCassity**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

VAProfileRedis::V2::ContactInformation needs to be configured for V2::ContactInformation to ensure user's vet360_contact_info is not affected by the upgrade in FormProfile and in mobile requests.

User's `vet360_contact_info` is dependent on VAProfileRedis::ContactInformation. To avoid any issues with mobile and prefill, `vaprofile_contact_info` and `va_profile_v2_email` replaces the exisiting `vet360_contact_info`. This will switch back after the upgrade is complete. 

The rails helper `stub_vaprofile_user` records the new VAProfileRedis::V2::ContactInformation information when obtaining user's vaprofile_contact_info for testing.

## Related issue(s)

- [issue](https://github.com/department-of-veterans-affairs/va.gov-team/issues/93641)

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: .github/CODEOWNERS
Comment (RachalCassity) [inline .github/CODEOWNERS:337]: app/models/va_profile_redis  directory covers this new file

File: app/models/va_profile_redis/contact_information.rb
Comment (RachalCassity) [inline app/models/va_profile_redis/contact_information.rb:3]: Removed v2 service form VAProfileRedis::ContactInformation

File: spec/models/va_profile_redis/cache_spec.rb
Comment (RachalCassity) [inline spec/models/va_profile_redis/cache_spec.rb:38]: Configured spec/models/va_profile_redis/v2/cache_spec.rb

File: spec/support/va_profile/stub_vet360.rb
Comment (RachalCassity) [inline spec/support/va_profile/stub_vet360.rb:2]: Configured spec/support/va_profile/stub_vaprofile_user.rb

**Review by LindseySaari — APPROVED** [review]

File: app/models/user.rb
Comment (bosawt) [inline app/models/user.rb:403]: @LindseySaari @RachalCassity I'd prefer if you all wait for Identity team approval when files that we own are changed.

Also, I might be misinterpreting based off the comments for these functions, but we really shouldn't be committing functions for spec testing in the production code.

Comment (RachalCassity) [reply app/models/user.rb:403]: Sorry, these are for the new service as well. User's vet360_contact_info is dependent on VAProfileRedis::ContactInformation. To avoid any issues with mobile and prefill, vaprofile_contact_info and va_profile_v2_email replaces the exisiting vet360_contact_info.

**Pull request merged**
